process/block/preprocess/basePreProcess.go

```diff
@@ -198,12 +198,15 @@ func (bpp *basePreProcess) computeExistingAndMissing(
 		}
 
 		if len(txHashes) > 0 {
-			missingTxsForShard[miniBlock.SenderShardID] = append(missingTxsForShard[miniBlock.SenderShardID],
-				&txsHashesInfo{txHashes: txHashes, receiverShardID: miniBlock.ReceiverShardID})
+			tmp := &txsHashesInfo{
+				txHashes:        sliceUtil.TrimSliceSliceByte(txHashes),
```

Comment:
instead of trim use index, instead of append can be put inside index. if the capacity is reached, double of current could be allocated. a small handler can be created for [][]byte type structures.

Reply:
for index vs append the rational is the same as above, 
"double of current could be allocated. a small handler can be created for [][]byte type structures." is in fact the definition of append

in current implementation only `txHashes` can ever grow and is doing it just to accommodate the maximum `len(miniBlock.TxHashes)` in all `len(body)`